Extensions/News/Compiler.js

```diff
@@ -0,0 +1,55 @@
+class News {
+    
+    /**
+     * 
+     * @param {HTMLElement} section 
+     */
+    constructor(section) {
+        this.section = section;
+        this.heading = null;
+        this.body = null;
+
+        this._parseHeading(this.section, (heading) => {
+            if (!heading) throw new Error("Nullptr found at [this._parseHeading] in class.News ctor<...>");
+            this.heading = heading;
+
+            this._parseBody(this.section, (body) => {
+                if (!body) throw new Error("Nullptr found at [this._parseHeading][this._parseBody] in class.News ctor<...>");
+                this.body = body;
+                delete this.section;
+            });
+        });
+
+        if (!this._initiated) throw new Error("There was an error while initializing class.News ctor<...>");
+    }
+
+    /**
+     * 
+     * @param {HTMLElement} section 
+     * @param {Function} result 
+     */
+    _parseHeading(section, callback) {
+        callback(section.getElementsByClassName("gadgetTitle")[0].textContent);
+    }
+
+    /**
+     * 
+     * @param {HTMLElement} section 
+     * @param {Function} result 
+     */
+    _parseBody(section, callback) {
+        callback(section.getElementsByClassName("plainText")[0].textContent);
+    }
+
+    /**
+     * 
+     * @returns {boolean}
+     */
+    _initiated() {
+        return (this.heading && this.body && (this.section == undefined)) ? true : false;
+    }
+}
+
+module.exports = {
+    News: News
+}
```

Comment:
very minor, but I thought we had a lint rule about newlines at end of file

Reply:
We didn't agree yet on whether to use ESLint, and if so then how to use it. Therefore, I didn't run any of the code through ESLint. We can talk about this tmrw.